src/common/base/src/ttl.rs

```diff
@@ -0,0 +1,182 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt::Display;
+use std::time::Duration;
+
+use serde::de::Visitor;
+use serde::{Deserialize, Serialize};
+
+/// Time To Live
+
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone, Copy, Default)]
+pub enum TimeToLive {
+    /// immediately throw away on insert
+    Immediate,
+    /// Duration to keep the data, this duration should be non-zero
+    Duration(Duration),
+    /// Keep the data forever
+    #[default]
+    Forever,
+    // TODO(discord9): add a new variant
+    // that can't be overridden by database level ttl? call it ForceForever?
```

Comment:
If table-level ttl is set to forever, it won't be overriden by database ttl.

Table level ttl should be an `Option<TimeToLive>`.

If it's None, then it follows database ttl. If database ttl is not set, then it's forever
If it's Some(*), it will always ignore database ttl.